src/package.lisp

```diff
@@ -28,6 +28,19 @@
       package-designator
       (oget *package-table* (string package-designator))))
 
+(defun package-designator-package (package-designator)
```

Comment:
There is some overlap between this function and `find-package` in `package.lisp`. 

`find-package` uses string to convert the package-designator into a name, and then get the package.  The only problem is that `string` is not signaling an error when it receives a non string designator, like a number.

If you use the typecase for string/symbol/character, so an error is signaled, then we can replace this function with just a call to `find-package-or-fail`.